src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2087,6 +2109,22 @@ private MethodSymbol InferMethod(BoundCall node, MethodSymbol method, ImmutableA
                 return definition.Construct(result.InferredTypeArguments);
             }
             return method;
+
+            BoundExpression makePlaceholder(TypeSymbolWithAnnotations argumentType, BoundExpression argument)
```

Comment:
>makePlaceholder [](start = 28, length = 15)

Can this be a static method a delegate for which we can cache and reuse?

Reply:
I'm not sure I understand the question. A local function without capture is already compiled into a static method.

(Example](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDZICYgNQA+AAgAwAERAjANwCwAUEQMwWUBsFATKwOxkDeDMsIosqHIgBZWADgAUASgFCRqlauHqNaAHYwyGAPYBjCBjm79EBAHMlAXgB8w6zbr1VAXwaegA===)

---
In reply to: [191574237](https://github.com/dotnet/roslyn/pull/26828#discussion_r191574237) [](ancestors = 191574237)

Reply:
The call-site (ZipAsArray) was indeed creating a new `Func` every time.

By having a static `Func` instead, this goes away.
Here's the result:
```
 // ImmutableArray<BoundExpression> arguments = ImmutableArrayExtensions.ZipAsArray<TypeSymbolWithAnnotations, BoundExpression, BoundExpression>(argumentTypes, node.Arguments, NullableWalker.makePlaceholder);
 IL_0014: ldarg.3
 IL_0015: ldarg.1
 IL_0016: callvirt instance valuetype [System.Collections.Immutable]System.Collections.Immutable.ImmutableArray`1<class Microsoft.CodeAnalysis.CSharp.BoundExpression> Microsoft.CodeAnalysis.CSharp.BoundCall::get_Arguments()
 IL_001b: ldsfld class [System.Runtime]System.Func`3<class Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbolWithAnnotations, class Microsoft.CodeAnalysis.CSharp.BoundExpression, class Microsoft.CodeAnalysis.CSharp.BoundExpression> Microsoft.CodeAnalysis.CSharp.NullableWalker::makePlaceholder
 IL_0020: call valuetype [System.Collections.Immutable]System.Collections.Immutable.ImmutableArray`1<!!2> [Microsoft.CodeAnalysis]Microsoft.CodeAnalysis.ImmutableArrayExtensions::ZipAsArray<class Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbolWithAnnotations, class Microsoft.CodeAnalysis.CSharp.BoundExpression, class Microsoft.CodeAnalysis.CSharp.BoundExpression>(valuetype [System.Collections.Immutable]System.Collections.Immutable.ImmutableArray`1<!!0>, valuetype [System.Collections.Immutable]System.Collections.Immutable.ImmutableArray`1<!!1>, class [System.Runtime]System.Func`3<!!0, !!1, !!2>)
```

Thanks!

---
In reply to: [191824709](https://github.com/dotnet/roslyn/pull/26828#discussion_r191824709) [](ancestors = 191824709,191574237)